src/pygeoprocessing/geoprocessing.py

```diff
@@ -1837,8 +1840,20 @@ def reclassify_raster(
             base_raster_path_band)
     raster_info = get_raster_info(base_raster_path_band[0])
     nodata = raster_info['nodata'][base_raster_path_band[1]-1]
-    keys = sorted(numpy.array(list(value_map.keys())))
-    values = numpy.array([value_map[x] for x in keys])
+    # If nodata was included in the value_map pop it from our lists
+    # and handle it separately. Doing this on the off chance nodata is
+    # a max or min float which can cause floating decimal discrepencies.
+    value_map_copy = value_map.copy()
+    nodata_dest_value = target_nodata
```

Comment:
What should happen if `target_nodata` is `None`? If `target_nodata` is `None` but there are nodata pixels in the input, we might end up introducing `nan`s into the output on line 1867. I think that case is invalid but should we check for it?

Reply:
Although, the docstring does say "numerical type". I'm actually more inclined to report an error if `target_nodata` is `None`.

Reply:
This is a really interesting question. As far as precedent, doesn't raster_calculator allow for the nodata value to be None? I think the source code for raster_calculator explicitly handles that case.

Reply:
> I think the source code for raster_calculator explicitly handles that case.

It does guard against an undefined target nodata value, you're right. I think this is a case where intuitively it doesn't make sense to me why you would pass in a None value for target nodata, however, if there is precedent of guarding against target nodata values of None, then we can certainly do it here as well.

`convolve_2d` at the very least handles None nodata by raising an error in a specific case, and setting to min float32 in another.